packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/NotificationTriggers.kt

```diff
@@ -0,0 +1,413 @@
+package expo.modules.notifications.notifications.triggers
+
+import android.os.Parcel
+import android.os.Parcelable
+import expo.modules.notifications.notifications.interfaces.NotificationTrigger
+import expo.modules.notifications.notifications.interfaces.SchedulableNotificationTrigger
+import java.io.Serializable
+import java.util.Calendar
+import java.util.Date
+
+open class ChannelAwareTrigger : NotificationTrigger, Serializable {
+  private var mChannelId: String?
+
+  constructor(channelId: String?) {
+    mChannelId = channelId
+  }
+
+  constructor(`in`: Parcel) {
+    mChannelId = `in`.readString()
+  }
+
+  override fun describeContents(): Int {
+    return 0
+  }
+
+  override fun writeToParcel(dest: Parcel, flags: Int) {
+    dest.writeString(mChannelId)
+  }
+
+  override fun getNotificationChannel(): String? {
+    return mChannelId
+  }
+
+  companion object {
+    @JvmField
+    val CREATOR: Parcelable.Creator<ChannelAwareTrigger?> =
+      object : Parcelable.Creator<ChannelAwareTrigger?> {
+        override fun createFromParcel(`in`: Parcel): ChannelAwareTrigger {
+          return ChannelAwareTrigger(`in`)
+        }
+
+        override fun newArray(size: Int): Array<ChannelAwareTrigger?> {
+          return arrayOfNulls(size)
+        }
+      }
+  }
```

Comment:
Can we use https://developer.android.com/kotlin/parcelize to simplify this code?

Reply:
Wow that would definitely make things nicer! I'll give that a try

Reply:
Ok I spent some time trying this, and it appears we would not be able to use this directly without writing custom parcelers as in https://developer.android.com/kotlin/parcelize#custom_parcelers .

I'll spend some more time to see if I can simplify this file further...

Reply:
Ok I figured out the problem I was having -- now using Parcelize and primary constructors.